atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,150 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import logging
+import json
+import paramiko
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class ResourceManager:
```

Comment:
@tveverka-RH  many class methods use the same variables in arguments. Why don't we use here instance variables (and @property decorator for some cases)?

I would revisit the structure Martin proposed earlier. 
```
RemoteHost(username, host, auth)
  * username
  * host
  * auth
  * check_avail()
  * unlock()
  * lock()
  ...

HostsPool()
  * hosts - list of RemoteHosts objects
  * load_from_config()
  * find()
  ....
```

Reply:
Only recurrence in the arguments I see only for procedure variables, instance variables are already in place. 

@Property decorator however is good point. -> going to use that